### PR TITLE
Publish

### PR DESCRIPTION
**Pull request by pampatzoglou**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,107 @@
+version: 2
+
+jobs:
+  requirements:
+    docker:
+      - image: web3f/ansible:v0.0.2
+    steps:
+      - checkout
+      - setup_remote_docker
```
Comment (fgimenez) [inline .circleci/config.yml]: No need for remote docker
```suggestion
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+      - run:
+          command: |
+            pip install -r w3f/substrate-benchmarks/requirements.txt
+
+  benchmark-1:
+    docker:
+      - image: web3f/ansible:v0.0.2
+    steps:
+      - checkout
+      - setup_remote_docker
```
Comment (fgimenez) [inline .circleci/config.yml]:
```suggestion
```

File: scripts/audit.sh [hunk 3]
```diff
@@ -0,0 +1 @@
+#!/bin/sh
```
Comment (fgimenez) [inline scripts/audit.sh]:
```suggestion
```

File: w3f/substrate-benchmarks/roles/gcp/vars/main.yml [hunk 4]
```diff
@@ -1,11 +1,11 @@
 ---
 # vars file for gcp
-desired_state: "{{ state | default('absent') }}"
-gcp_project:  "{{ set_gcp_project | default('polkadot-benchmarks') }}"
+gcp_project:  "{{ lookup('env', 'GCP_PROJECT') | default('polkadot-benchmarks') }}"
+gcp_cred_file: "{{ lookup('env', 'GOOGLE_APPLICATION_CREDENTIALS_CONTENT') }}"
```
Comment (fgimenez) [inline w3f/substrate-benchmarks/roles/gcp/vars/main.yml]: `gcp_cred_file` should contain the path to the credentials file and the `GOOGLE_APPLICATION_CREDENTIALS_CONTENT` env var has the contents of the credentials file. You should previously store this contents on a file with the path stored in `GOOGLE_APPLICATION_CREDENTIALS` and set that variable instead here.

File: .circleci/config.yml [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+          requires:
+            - requirements
+      - benchmark-2:
+          filters:
+            branches:
+              ignore: /.*/
+            tags:
+              only: /v[0-9]+(\.[0-9]+)*/
+          requires:
+            - benchmark-1
```
Comment (fgimenez) [inline .circleci/config.yml:86]: Why not running all the benchmarks in parallel?
```suggestion
            - requirements
```

File: .circleci/config.yml [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+      - benchmark-3:
+          filters:
+            branches:
+              ignore: /.*/
+            tags:
+              only: /v[0-9]+(\.[0-9]+)*/
+          requires:
+            - benchmark-2
```
Comment (fgimenez) [inline .circleci/config.yml]:
```suggestion
            - requirements
```

File: .circleci/config.yml [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+      - benchmark-4:
+          filters:
+            branches:
+              ignore: /.*/
+            tags:
+              only: /v[0-9]+(\.[0-9]+)*/
+          requires:
+            - benchmark-3
```
Comment (fgimenez) [inline .circleci/config.yml]:
```suggestion
            - requirements
```

**Review by fgimenez — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 5 above)
Comment (pampatzoglou) [reply .circleci/config.yml:86]: bc we use the same compute instance name. I will refactor.

**Pull request closed without merging**